books/RayTracingTheNextWeek.html

```diff
@@ -2256,7 +2256,319 @@
+origin to any point on the plane), then
+
+  $$ \mathbf{n} \cdot \mathbf{v} = D $$
+
+Now to find the intersection with some ray $\mathbf{R}(t) = \mathbf{P} + t\mathbf{d}$. Plugging in
+the ray equation, we get
+
+  $$ \mathbf{n} \cdot ( \mathbf{P} + t \mathbf{d} ) = D $$
+
+Solving for $t$:
+
+  $$ \mathbf{n} \cdot \mathbf{P} + \mathbf{n} \cdot t \mathbf{d}  = D $$
+
+  $$ \mathbf{n} \cdot \mathbf{P} + t(\mathbf{n} \cdot \mathbf{d}) = D $$
+
+  $$ t = \frac{D - \mathbf{n} \cdot \mathbf{P}}{\mathbf{n} \cdot \mathbf{d}} $$
+
+This gives us $t$, which we can plug into the ray equation to find the point of intersection.
+
+Ok, we can find the point of intersection between a ray and a plane. This will help us determine
+whether a ray hits a given quadrilateral, using the plane that contains the (planar) quadrilateral.
+In fact, we can use this approach to define any planar primitive, like triangles and disks (more
+later).
+
+We now have three questions we must answer:
+
+1. How do we define the quadrilateral in three dimensional space?
+2. How do we find the plane that contains that quadrilateral?
+3. How do we orient the ray-plane intersection point within the quadrilateral?
+
+
+Defining the Quadrilateral
+---------------------------
+For our purposes, we'll use three geometric entities to define a quadrilateral. Remember that when
+we say "quadrilateral" here, we're really talking about parallelograms, a specific kind of
+quadrilateral where opposite sides are parallel. These three entities are
+
+1. $\mathbf{O}$, the lower-left corner of the quadrilateral.
+2. $\mathbf{u}$, a vector representing the first side of the quadrilateral.
+   $\mathbf{O} + \mathbf{u}$ gives the lower right corner.
+3. $\mathbf{v}$, a vector representing the second side of the quadrilateral.
+   $\mathbf{O} + \mathbf{v}$ gives the upper left corner.
+
+These values are three-dimensional, even though a quad itself is a two-dimensional object.
+
+The following figure illustrates the quadrilateral components.
+
+  ![Figure [quad-def]: Quadrilateral Components](../images/fig-2.05-quad-def.jpg)
+
+
+Finding the Plane That Contains a Given Quadrilateral
+------------------------------------------------------
+We have $O$, $u$, and $v$, and want the corresponding equation of the plane containing all three of
+these values (and thus the quad itself).
+
+Fortunately, this is very simple. Recall that in $Ax + By + Cz = D$, $(A,B,C)$ represents the
+normal vector. To get this, we just use the cross product of the two side vectors $u$ and $v$:
+
+  $$ \mathbf{n} = \text{unit_vector}(\mathbf{u} \times \mathbf{v}) $$
+
+The plane is defined as all points $(x,y,z)$ that satisfy the equation $Ax + By + Cz = D$. Well, we
+know that $\mathbf{O}$ lies on the plane, so we can use that to find $D$:
+
+  $$ D = \mathbf{n}_x\mathbf{O}_x + \mathbf{n}_y\mathbf{O}_y + \mathbf{n}_z\mathbf{O}_z $$
```

Comment:
$$ D = n \cdot O $$
  $$ D = \mathbf{n}_x\mathbf{O}_x + \mathbf{n}_y\mathbf{O}_y + \mathbf{n}_z\mathbf{O}_z $$

Reply:
Actually, I think it reads better in the opposite order, so I did that.